Export LLM documentation revamp (#12381)

jackzhxng opened the pull request:
Merge https://github.com/pytorch/executorch/pull/12381 from side branch into main.

jackzhxng committed:
Export LLM documentation revamp (#12381)

mergennachin commented:
Incorporate these two PRs? 

https://github.com/pytorch/executorch/pull/12802
https://github.com/pytorch/executorch/pull/12815

mergennachin committed:
[Release 0.7]: Fix lint (#12815)

Gasoonjia committed:
update export llm doc (#12802)

This PR update model profiling doc in Exporting custom LLMs and
Exporting LLMs. Specially:

1. removed export-irrevalnt profiling settings and progress (etdump
generation and inspector usage) in export doc
2. reformat the log output for better demonstration
3. replace the link from absolute link to relateive one for version
control
4. other small updates

jackzhxng committed:
Merge branch 'main' into export-llm-docs

jackzhxng committed:
Lint
